### DISCUSSION TITLE
Faster searching strategies?

### DISCUSSION THREAD

**Discussion by Ch0ronomato** [BurntSushi/ripgrep#1985] [Q&A] [answered]
Hi!

Just curious - I'm doing a dummy search over a cidr range `10.11.128.0/17`, which I wrote as

```
for ip in 10.11.{128..255}.{0..255}
do
rg -F $ip
done
```

it's not that **slow**, but it is pretty beefy and uses a lot of resources - any suggestions on how to improve the peformance?

It searches over these directories btw
```
 18M	dir1
1000M	dir2
```

**Comment by BurntSushi** [comment, marked as answer]
Without a full reproduction, it's hard to say, but why not ask ripgrep to search for all things simultaneously? e.g.,

```
$ for ip in 10.11.{128..255}.{0..255}; do
  echo $ip >> patterns
done
$ rg -F -f patterns
```
